pkg/ddc/alluxio/ufs_internal.go

```diff
@@ -166,35 +166,26 @@ func (e *AlluxioEngine) processUpdatingUFS(ufsToUpdate *utils.UFSToUpdate) (err
 		alluxioPath := utils.UFSPathBuilder{}.GenAlluxioMountPath(mount, dataset.Spec.Mounts)
 		if len(ufsToUpdate.ToAdd()) > 0 && utils.ContainsString(ufsToUpdate.ToAdd(), alluxioPath) {
 			mountOptions := map[string]string{}
+			for key, value := range dataset.Spec.SharedOptions {
+				mountOptions[key] = value
+			}
+
 			for key, value := range mount.Options {
 				mountOptions[key] = value
 			}
 
 			// Configure mountOptions using encryptOptions
 			// If encryptOptions have the same key with options, it will overwrite the corresponding value
-			for _, encryptOption := range mount.EncryptOptions {
-				key := encryptOption.Name
-				// Add senstive key filter
-				securityutil.UpdateSensitiveKey(key)
-				secretKeyRef := encryptOption.ValueFrom.SecretKeyRef
-
-				secret, err := kubeclient.GetSecret(e.Client, secretKeyRef.Name, e.namespace)
-				if err != nil {
-					e.Log.Info("can't get the secret",
-						"namespace", e.namespace,
-						"name", e.name,
-						"secretName", secretKeyRef.Name)
-					return err
-				}
-
-				value := secret.Data[secretKeyRef.Key]
-				e.Log.Info("get value from secret",
-					"namespace", e.namespace,
-					"name", e.name,
-					"secretName", secretKeyRef.Name)
-
-				mountOptions[key] = string(value)
+			mountOptions, err = e.genEncryptOptions(dataset.Spec.SharedEncryptOptions, mountOptions, mount.Name)
+			if err != nil {
+				return err
```

Comment:
I think it's better to swap the overriding order for `SharedEncryptOptions` and the `mount.Options`. In my opinion, `mount.Options` should override `SharedEncryptOptions`

Reply:
I think mount.Options and  EncryptOptions is different, There is no override relationship between them.

Reply:
How do we handle this if the key of mount.Options and EncryptOptions is  the same? Should we throw error out?

Reply:
I think  throw error out is ok.

Reply:
Please optimize in next PR. thanks.

Reply:
ok